subworkflows/nf-core/fastq_align_dedup_bismark/main.nf

```diff
@@ -0,0 +1,154 @@
+include { BISMARK_ALIGN                                 } from '../../../modules/nf-core/bismark/align/main'
+include { SAMTOOLS_SORT as SAMTOOLS_SORT_DEDUPLICATED   } from '../../../modules/nf-core/samtools/sort/main'
+include { SAMTOOLS_INDEX as SAMTOOLS_INDEX_DEDUPLICATED } from '../../../modules/nf-core/samtools/index/main'
+include { BISMARK_DEDUPLICATE                           } from '../../../modules/nf-core/bismark/deduplicate/main'
+include { BISMARK_METHYLATIONEXTRACTOR                  } from '../../../modules/nf-core/bismark/methylationextractor/main'
+include { BISMARK_COVERAGE2CYTOSINE                     } from '../../../modules/nf-core/bismark/coverage2cytosine/main'
+include { BISMARK_REPORT                                } from '../../../modules/nf-core/bismark/report/main'
+include { BISMARK_SUMMARY                               } from '../../../modules/nf-core/bismark/summary/main'
+
+workflow FASTQ_ALIGN_DEDUP_BISMARK {
+
+    take:
+    ch_reads             // channel: [ val(meta), [ reads ] ]
+    ch_fasta             // channel: [ val(meta), [ fasta ] ]
+    ch_bismark_index     // channel: [ val(meta), [ bismark index ] ]
+    skip_deduplication   // boolean: whether to deduplicate alignments
+    cytosine_report      // boolean: whether the run coverage2cytosine
+
+    main:
+    ch_versions                   = Channel.empty()
+    ch_coverage2cytosine_coverage = Channel.empty()
+    ch_coverage2cytosine_report   = Channel.empty()
+    ch_coverage2cytosine_summary  = Channel.empty()
+    ch_methylation_bedgraph       = Channel.empty()
+    ch_methylation_calls          = Channel.empty()
+    ch_methylation_coverage       = Channel.empty()
+    ch_methylation_report         = Channel.empty()
+    ch_methylation_mbias          = Channel.empty()
+    ch_bismark_report             = Channel.empty()
+    ch_bismark_summary            = Channel.empty()
+    ch_multiqc_files              = Channel.empty()
+
+    /*
+     * Align with bismark
+     */
+    BISMARK_ALIGN (
+        ch_reads,
+        ch_fasta,
+        ch_bismark_index
+    )
+    ch_versions = ch_versions.mix(BISMARK_ALIGN.out.versions)
```

Comment:
```suggestion
    ch_versions = ch_versions.mix(BISMARK_ALIGN.out.versions.first())
```
I'm not sure it matters as duplicates are filtered out in the end anyway so it's up to you if you want to use `.first()` on all the module version outputs.

Reply:
good point. will apply

Reply:
and you are right again that the test doesn't need that extensive config options. removed as applicable for the test

Reply:
going to leave it out for now. Will add back if convinced it has utility